nexus/src/app/mod.rs

```diff
@@ -105,7 +106,10 @@ pub struct Nexus {
     // this amount of time is called "max issue delay" and we have to set that
     // in order for our integration tests that POST static SAML responses to
     // Nexus to not all fail.
-    samael_max_issue_delay: std::sync::Mutex<Option<chrono::Duration>>,
+    samael_max_issue_delay: Mutex<Option<chrono::Duration>>,
+
+    pub resolver:
+        Arc<tokio::sync::Mutex<internal_dns_client::multiclient::Resolver>>,
```

Comment:
IMO it's questionable to expose a `pub` Mutex; this is easy to use incorrectly

Reply:
yeah, fixed in 1bd3ff98